lib/ext/heroku/command/addons.rb

```diff
@@ -341,25 +344,27 @@ def destroy
       ).body.keep_if do |attachment|
         attachment['addon']['name'] == addon
       end
-      config_vars = addon_attachments.map { |attachment| attachment['name'] }
 
-      config_vars.each do |var_name|
-        action("Removing #{addon} as #{var_name} from #{app}") {}
-        action("Unsetting #{var_name} and restarting #{app}") {}
+      attachment_names = addon_attachments.map { |attachment| attachment['name'] }
+      attachment_names.each do |attachment_name|
+        action("Removing #{addon} as #{attachment_name} from #{app}") {}
       end
 
-      @status = api.get_release(app, 'current').body['name']
-      action("Destroying #{addon} on #{app}") do
-        api.request(
-          :body     => json_encode({
-            "force" => options[:force],
-          }),
-          :expects  => 200..300,
-          :headers  => { "Accept" => "application/vnd.heroku+json; version=edge" },
-          :method   => :delete,
-          :path     => "/apps/#{app}/addons/#{addon}"
-        )
+      destroyed_addon = api.request(
+        :body     => json_encode({
+          "force" => options[:force],
+        }),
+        :expects  => 200..300,
+        :headers  => { "Accept" => "application/vnd.heroku+json; version=edge" },
+        :method   => :delete,
+        :path     => "/apps/#{app}/addons/#{addon}"
+      ).body
+
+      if destroyed_addon['config_vars'].any?
```

Comment:
Note, `addon['config_vars']` as a field only exists for historical reasons (it used to represent the attachment) and so it only lists vars that were on the billing app (from any billing app attachment). Ideally, we'd see all the vars on all the attachments and output the vars that have been removed from each app.

This will work correctly in the non-`--force` case, because the removal won't go through when there are many apps attached. With `--force`, the output below will be incomplete and multiple apps might have been restarted.